app/services/logging/logger.js

```diff
@@ -1,17 +1,143 @@
+      this.loggerConfig.path &&
+      this.loggerConfig.fileName
+    ) {
+      transports.push(
+        new this.winston.transports.DailyRotateFile({
+          // Check if the provided filename has the rolling
+          // date placeholder - if not, append it.
+          filename: /%DATE%/g.test(this.loggerConfig.fileName)
+            ? this.loggerConfig.fileName
+            : this.loggerConfig.fileName + "%DATE%",
+          dirname: this.loggerConfig.path,
+          datePattern: this.loggerConfig.rotate.datePattern,
+          zippedArchive: this.loggerConfig.rotate.zipped,
+          maxSize: this.loggerConfig.rotate.maxSize,
+          maxFiles: this.loggerConfig.rotate.maxFiles,
+          format: this.getLogFormat(),
+        })
+      );
+    } else if (
+      this.hasTransport(LoggerTransports.FILE) &&
+      this.loggerConfig.path &&
+      this.loggerConfig.fileName
+    ) {
+      transports.push(
+        new this.winston.transports.File({
+          // If a rolling date placeholder has somehow crept
+          // through, let's make sure to remove it here.
+          filename: path.join(
+            this.loggerConfig.path,
+            this.loggerConfig.fileName.replace(/%DATE%/g, "")
+          ),
+          format: this.getLogFormat(),
+        })
+      );
+    }
+
+    if (transports.length) {
+      this.logger = this.winston.createLogger({
+        level: this.loggerConfig.level,
+        transports,
+      });
+    }
+
+    return this;
+  }
+
+  exec(level, message, optionalParams) {
+    if (!level) {
+      level = "info";
+    }
+
+    // Fall back to default console implementation if
+    // no transports are specified.
+    if (!this.logger) {
+      if (optionalParams && optionalParams.length) {
+        console[level](message, optionalParams);
+      } else {
+        console[level](message);
+      }
+      return;
+    }
+
     if (optionalParams && optionalParams.length > 0) {
-      console.log(message, optionalParams);
+      this.logger.log({
+        level,
+        message,
+        meta: optionalParams,
+      });
     } else {
-      console.log(message);
+      this.logger.log({ level, message });
     }
   }
-}
+
+  log(message, ...optionalParams) {
+    this.exec("info", message, optionalParams);
+  }
+
+  debug(message, ...optionalParams) {
+    this.exec("debug", message, optionalParams);
+  }
+
+  info(message, ...optionalParams) {
+    this.exec("info", message, optionalParams);
+  }
+
+  warn(message, ...optionalParams) {
+    this.exec("warn", message, optionalParams);
+  }
+
+  error(message, ...optionalParams) {
+    this.exec("error", message, optionalParams);
+  }
+
+  getLogFormat() {
+    return this.winston.format.combine(
+      this.winston.format.timestamp({
+        format: this.loggerConfig.timestampFormat,
+      }),
+      this.winston.format.printf(
+        (info) => `${info.timestamp} [${info.level}] ${info.message}`
+      )
+    );
+  }
+
+  hasTransport(transport) {
```

Comment:
This is a great example of how to write well maintainable code - a small method, with one use, using return statements to control the flow.